modules/caddyhttp/reverseproxy/caddyfile.go

```diff
@@ -1397,8 +1398,35 @@ func (u *AUpstreams) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
 				}
 				u.FallbackDelay = caddy.Duration(dur)
 
+			case "versions":
+				args := d.RemainingArgs()
+				if len(args) == 0 {
+					return d.Errf("Must specify at least one version")
+				}
+
+				// Set up defaults; if versions are specified,
+				// both versions start as false, then flipped
+				// to true otherwise.
+				falseBool := false
+				u.Versions = ipVersions{
+					IPv4: &falseBool,
+					IPv6: &falseBool,
+				}
```

Comment:
Doing this, the one that isn't in the Caddyfile config still appears in the JSON output. I think this should only set the version to true if it appears in the config.

Essentially, you should do:
```
if u.Versions == nil {
	u.Versions = ipVersions{}
}
```

Which leaves them nil, and then in the for-switch just set entries to true.

Reply:
woops, my bad.
Thanks for clarifying!


Will do